### PR TITLE
Pea Plants

### PR DESCRIPTION
**Pull request by ZeroDayDaemon**
## About the PR

In this PR I have added pea plants and made pea soup craftable/cookable with them, and added the seeds to vendors and crates

## Why / Balance

Pea plants were missing.
Additionally, peas can now be used to make pea soup.

## Technical details

This is just YAML work.

## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Daemon
- add: Add pea plants.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
make the peas turn into pea shooters when given unstable mutagen

it would be like a turret that shoots at nuke ops and zombies

**Comment by drteaspoon420** [comment]
Might be non-issue, but default `PotencyDivisor` in all other plants is based on `100/Max`.
so for 4 -> 25 and for 3 -> 33

**Comment by Partmedia** [comment]
I don't see any duck updates, but the PR description indicates that they should be there. Per the [PR guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html), please keep unrelated changes in separate PR's. If you are finished with the pea plant updates, please update the PR title/description and undraft.

**Comment by ZeroDayDaemon** [comment]
> I don't see any duck updates, but the PR description indicates that they should be there. Per the [PR guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html), please keep unrelated changes in separate PR's. If you are finished with the pea plant updates, please update the PR title/description and undraft.

The duck related content has not been finished yet, it was mentioned because it was planned and this PR is still in draft mode.
Duck related content was considered relevant to me as ducks love peas, but I am willing to do the duck related stuff in another PR.

**Comment by ZeroDayDaemon** [comment]
> Might be non-issue, but default `PotencyDivisor` in all other plants is based on `100/Max`. so for 4 -> 25 and for 3 -> 33

Yeah i had NFI what that was or what it means lol, will fix with that

**Comment by drteaspoon420** [comment]
> > Might be non-issue, but default `PotencyDivisor` in all other plants is based on `100/Max`. so for 4 -> 25 and for 3 -> 33
> 
> Yeah i had NFI what that was or what it means lol, will fix with that

To be honest, I was not exactly sure about the potency divisor, just noticed the trend when working on the mutation system.
Just now added some documentation about the potency divisor (in a PR #20551 ).
> When chemicals are added to produce, the potency of the seed is divided with this value. Final chemical amount is the result plus the `Min` value.
> Example: PotencyDivisor of 20 with seed potency of 55 results in 2.75, 55/20 = 2.75. If minimum is 1 then final result will be 3.75 of that chemical, 55/20+1 = 3.75.

**Comment by Doru991** [comment]
The sprites for the pea plant in the tray are too low I think, due to the way plants are rendered there should be (as a rule of thumb) about 8 pixels between the bottom of the image and the lowest pixel in the sprite

**Comment by deltanedas** [comment]
yeah they are definitely too low they should grow in the tray not at the bottom edge of the tile

**Comment by ZeroDayDaemon** [comment]
Pea plants are a vine, so it makes sense for them to be growing off the side of the hydroponics tray, although they possibly could be moved up a few pixels

**Comment by ZeroDayDaemon** [comment]
Yeah, after trying a few variations, moving them up without making them look messed up would require different textures. Pea plant textures were pulled from /tg/ and as far as I know, share the same hydroponics tray texture.

File: Resources/Prototypes/Entities/Objects/Specific/Hydroponics/seeds.yml [hunk 1]
```diff
@@ -497,4 +497,15 @@
     - type: Seed
       seedId: bungo
     - type: Sprite
-      sprite: Objects/Specific/Hydroponics/bungo.rsi
\ No newline at end of file
+      sprite: Objects/Specific/Hydroponics/bungo.rsi
+
+- type: entity
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Objects/Specific/Hydroponics/seeds.yml:502]: Please double check your [YAML conventions](https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#yaml-conventions)

File: Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml [hunk 2]
```diff
@@ -1565,3 +1565,26 @@
     seedId: bungo
   - type: SpaceGarbage
   - type: BadFood
+
+- type: entity
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml:1569]: Please double check your [YAML conventions](https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#yaml-conventions)

**Review by Partmedia — APPROVED** [review]
Besides some minor nitpicks that I hope the submitter will address, this looks fine.

File: Resources/Prototypes/Entities/Objects/Specific/Hydroponics/seeds.yml (hunk 1 above)
Comment (ZeroDayDaemon) [reply Resources/Prototypes/Entities/Objects/Specific/Hydroponics/seeds.yml:502]: I have changed the order to be parent, id, name and then description.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml (hunk 2 above)
Comment (ZeroDayDaemon) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml:1569]: I have changed the order to be parent, id, name and then description.

**Comment by ZeroDayDaemon** [comment]
WHAT THE FUCK

**Comment by Carolyn3114** [comment]
is it possible to create a “pea turret” mutation that just shoots peas that work like pies and stun people?

**Comment by ZeroDayDaemon** [comment]
> is it possible to create a “pea turret” mutation that just shoots peas that work like pies and stun people?

Its possible yes (I probably wont do it)

**Comment by mirrorcult** [comment]
that is definitely possible and also the most based idea ive heard

**Pull request merged**